spec/lib/common/connection-spec.js

```diff
@@ -5,113 +6,254 @@
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                return expect(startPromise).to.be.rejectedWith(/Exceeded max retries/);
+            it('should suppress Error Code ECONNRESET', function () {
+                amqp.createConnection = sinon.spy(function () {
+                    return ({
+                        'on': function (a, callback) {
+                            callback({'code': 'ECONNRESET'}, {});
+                        }
+                    })
+                });
+                return this.subject.start();
             });
 
-            it('should increment retry attempts if initial connection fails', function() {
-                this.subject.initialConnection = false;
-                this.subject.start();
-                this.subject.connection.removeAllListeners('ready');
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                expect(this.subject.initialConnectionRetries).to.equal(3);
+            it('should check if initialConnection is true', function () {
+                this.subject.initialConnection = {};
+                amqp.createConnection = sinon.spy(function () {
+                    return ({
+                        'on': function (a, callback) {
+                            callback({'message': 'Hit an error'}, {});
+                        }
+                    })
+                })
+                return this.subject.start();
             });
 
-            it('should not increment retry attempts if initial connection succeeded', function() {
-                this.subject.initialConnection = true;
-                this.subject.start();
-                this.subject.connection.removeAllListeners('ready');
-                this.subject.connection.emit('error', new Error('test'));
-                this.subject.connection.emit('error', new Error('test'));
-                expect(this.subject.initialConnectionRetries).to.equal(0);
+            it('should check if max retries exceeded', function () {
+                this.subject.initialConnectionRetries = 100;
+                amqp.createConnection = sinon.spy(function () {
+                    return ({
+                        'on': function (a, callback) {
+                            if (a === 'error') {
+                                callback({'message': 'error'}, {});
+                            }
+                        }
+                    })
+                })
+                return this.subject.start()
+                    .catch(function (Error) {
+                        console.log(Error);
```

Comment:
I don't think we would want to catch here, otherwise the test will pass on start() failures.